matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/room/membership/DefaultMembershipService.kt

```diff
@@ -127,6 +131,10 @@ internal class DefaultMembershipService @AssistedInject constructor(
     }
 
     override suspend fun invite(userId: String, reason: String?) {
+        val sessionInfoSet = Realm.getInstance(monarchy.realmConfiguration).use {
```

Comment:
should we only share the keys if the invite succeeds?

Reply:
No we should share them even if the invite fails

Reply:
Yes its a good point. Though, we have already discussed it with the crypto team and sharing prior to a user invite is accepted. It has more pros than cons like some performance benefits. Also keep in mind that the proper/robust implementation for that feature will be implemented through the rust SDK. So it's not a permanent solution.